### DISCUSSION TITLE
Registration Disabled Error on Login Attempt

### DISCUSSION THREAD

**Discussion by alpeer** [verdaccio/verdaccio#4858] [Q&A] [answered]
Problem: It returns E409 on Login, not sure if it's a bug

I disabled registration by setting `max_users: -1` and added user manually to htpasswd (I tested login before disabling registration and login/publishing was working properly)
After disabling registration I tried login by `npm login --registry=https://MY_VERDACCIO_ADDRESS_MASKED` and entered credentials.
It responded like this:
```
npm notice Log in on https://MY_VERDACCIO_ADDRESS_MASKED
Username: admin
Password: 

npm error code E409
npm error 409 Conflict - PUT https://MY_VERDACCIO_ADDRESS_MASKED/-/user/org.couchdb.user:admin/-rev/undefined - user registration disabled
npm error A complete log of this run can be found in: /Users/*********/.npm/_logs/2024-09-23T22_19_13_185Z-debug-0.log
```

Here's my part of verdaccio config
```yaml
  htpasswd:
    file: ./htpasswd
    max_users: -1
    algorithm: bcrypt
```

---
What I'm trying to do is allowing logins only from manually added ones(which is on htpasswd) and disable registration.
I tried `max_users: 0` before and it was allowing registration by `npm adduser` command, which I don't want to allow.
Any other solutions welcomed.

**Comment by CleyFaye** [comment, marked as answer]
That's quite an old question, but I have a potential answer. On a new installation, the DB file is not created. The DB file is where the "secret" for auth is stored.

On failed login attempts, this file is not created. And trying to login when this secret does not exist fails, so it will never be created.

As a dirty fix, setting `max_users: 1`, doing a login with an user from the `htpasswd` file, and reverting to `max_users: -1` generate the secret and works without issues afterward.

**Comment by mbtools** [comment]
You have one existing user, so `max_users: 1` should be the correct setting preventing registration of additional users.
